dartagnan/src/main/java/com/dat3m/dartagnan/program/analysis/alias/InclusionBasedPointerAnalysis.java

```diff
@@ -842,9 +842,15 @@ yield new Result(null, null, BigInteger.ZERO,
 
         @Override
         public Result visitIntUnaryExpression(IntUnaryExpr x) {
+            if (x.getKind() != IntUnaryOp.MINUS) {
+                return null;
+            }
             final Result result = x.getOperand().accept(this);
-            return result == null ? null : x.getKind() != IntUnaryOp.MINUS ? result :
-                    new Result(null, null, result.offset.negate(), result.alignment.isEmpty() ? TOP : result.alignment);
+            final var alignment = new ArrayList<Integer>();
+            for (final Integer a : result.alignment) {
+                alignment.add(-Math.abs(a));
+            }
+            return new Result(null, null, result.offset.negate(), alignment);
```

Comment:
This is not correct for `-r`, no? It seems that you would return the constant `-0=0` rather than `0 + -1*x`.
I think whenever you have some base `b`, then this base should be considered 1-aligned when doing negation.